src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -346,28 +376,37 @@ export class DataSourceSavedObjectsClientWrapper {
             'Update failed due to deprecated data source: "auth.credentials.username" missing. Please delete and create another data source.'
           );
         }
+        return this.validateEncryptionContext(password, endpoint);
+      case AuthType.SigV4:
+        const { accessKey } = auth.credential as SigV4Content;
+        return this.validateEncryptionContext(accessKey, endpoint);
+      default:
+        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${auth.type}'`);
+    }
+  }
 
-        const { encryptionContext } = await this.cryptography
-          .decodeAndDecrypt(password)
-          .catch((err: any) => {
-            const errMsg = `Failed to update existing data source for dataSourceId [${id}]: unable to decrypt "auth.credentials.password"`;
-            this.logger.error(errMsg);
-            this.logger.error(err);
-            throw SavedObjectsErrorHelpers.decorateBadRequestError(err, errMsg);
-          });
+  private async validateEncryptionContext(input: string, reference: string) {
+    const { encryptionContext } = await this.cryptography
+      .decodeAndDecrypt(input)
+      .catch((err: any) => {
+        const errMsg = `Failed to update existing data source: unable to decrypt "auth.credentials.password"`;
+        this.logger.error(errMsg);
+        this.logger.error(err);
+        throw SavedObjectsErrorHelpers.decorateBadRequestError(err, errMsg);
+      });
 
-        if (encryptionContext.endpoint !== endpoint) {
-          throw SavedObjectsErrorHelpers.createBadRequestError(
-            'Update failed due to deprecated data source: "endpoint" contaminated. Please delete and create another data source.'
-          );
-        }
-        return encryptionContext;
-      default:
-        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${type}'`);
+    if (encryptionContext.endpoint !== reference) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        'Update failed due to deprecated data source: "endpoint" contaminated. Please delete and create another data source.'
```

Comment:
What do we mean by "deprecated data source"?

Reply:
+1

Reply:
This is from previous contributor. update the text to be just `Failed to update datasource: xxx`